src/_zkapauthorizer/tests/matchers.py

```diff
@@ -140,8 +140,9 @@ def leases_current(relevant_storage_indexes, now, min_lease_remaining):
         # visited and maintained.
         lambda storage_server: list(
             stat
-            for (storage_index, stat) in storage_server.buckets.items()
+            for (storage_index, shares) in storage_server.buckets.items()
             if storage_index in relevant_storage_indexes
+            for (sharenum, stat) in shares.items()
```

Comment:
I find this difficult to understand.
It is not a list comprehension, but something similar;
it lives in a list constructor.
It has no indentation.

list(stat for (tuple) in (list) if (relevant) for (shares) ... ❓

Can you give me a pointer so I can RTFM?

Reply:
Yea this syntax sucks.  I'll rewrite it using a for loop statement instead.  For your reference, there's https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions but I don't know if it makes any sense to someone who doesn't already know everything it is trying to say.

This *is* a list comprehension - with two loops in it.  The outer loop is the first `for` and has a filtering condition that applies to the inner loop which is the second `for`.  The elements that end up in the list are the values of the expression at the beginning, as usual, evaluated once for each iteration of the inner loop within each iteration of the outer loop.